### ISSUE TITLE
Please integrate mptcp into kernel LTS.

### ISSUE THREAD

**Issue by enihcam** [multipath-tcp/mptcp#139]
Please integrate mptcp into kernel LTS.

**Comment by madhanraj** [comment]
any updates on the same ?

**Comment by cpaasch** [comment]
If it would be that easy, it would already have been done ;)

**Comment by madhanraj** [comment]
Working on latest MPTCP version in 3.18.14 Linux Kernel

**Comment by okias** [comment]
Maybe would be cool publish some list of "TODO", what should be done before _asking_ for inclusion in mainline...

Missing MPTCP really bugs me, but I can't do nothing about it.. (or at least watch some TODO list how it moving forward)

**Comment by pRiVi** [comment]
I am VERY VERY happy that such unstable patches like this are not getting into mainline, as you can see on #130 and #106.

There must be much investigation for searching such issues, like memory corruption. Most times it is much more easy to reimplement once again, making things this time more robust and with better and more testing from step to step.

**Comment by okias** [comment]
@pRiVi Relax, MPTCP needs more developers and sadly I guess it's not going happen until it hits mainline kernel or at least there will be live discussion about including in mainline.

Also @cpaasch , you're still working fulltime in Apple, right?

**Comment by cpaasch** [comment]
@okias Yes, still fulltime at Apple :) But, getting more time to spend on MPTCP now ;)

Wrt. to the ToDo-list, that's a good point! I will make one at https://multipath-tcp.org/pmwiki.php/Developer/Upstreaming

**Comment by cpaasch** [comment]
I also tag these ToDo-items with the label "For Upstreaming" https://github.com/multipath-tcp/mptcp/labels/For%20Upstreaming

**Comment by wrouesnel** [comment]
Building out the main git kernel I'm noticing it's at 4.1. Is there any word on the relative difficulty of porting forward the patches to more recent kernels (4.4 - 4.8 ?)

**Comment by cpaasch** [comment]
mptcp_trunk is currently based on v4.4

**Comment by nvsystems** [comment]
Just wanted to start using mptcp and was astounded that there are no patches available for the current 4.9 longterm stable kernel from kernel.org - kernel 4.4 ist just WAY to old for me to use :-(
(one of many reasons: btrfs improvements since 4.4, but also other recent changes to the kernel that I need)

Would be nice if mptcp gets merged into the mainline kernel... I'm very curious to try it, but as long as it is not in the mainline kernel it is really hard to use.

**Comment by enihcam** [comment]
If integrating MPTCP into kernel mainline is difficult, can we just align the patch with kernel LTS?

**Comment by cpaasch** [comment]
MPTCP is aligned with the LTS in branch mptcp_v0.92 (current stable release based on v4.4.x) and mptcp_v0.93 (upcoming stable release, based on v4.9.x). mptcp_v0.93 currently has still a bug that I am working on.

**Comment by enihcam** [comment]
@cpaasch, brilliant! I have been waiting for a long time and looking forward to v4.9.x with MPTCP, because since v4.9 BBR (congestion control algorithm from Google) is introduced.

**Comment by cypa** [comment]
@cpaasch what is the bug exactly, please? (v0.93 bug)

**Comment by cpaasch** [comment]
The bug was related to ofo-queue processing, but has been fixed in the latest release.
